samples/interactive-execute-context/src/executeTemplate.ts

```diff
@@ -0,0 +1,82 @@
+/*
+syntax enhancement for the demo - temporary
+*/
+import { MagickFile, isImage as isImageFile } from 'wasm-imagemagick'
+import { ExecuteConfig } from 'imagemagick-browser'
+
+export interface CommandContext {
+  commands: string[][],
+  files: MagickFile[],
+  defaultImage?: MagickFile
+  placeholders?: CommandContextPlaceholders
+}
+export interface CommandContextPlaceholders {
+  singleImagePrefix?: string,
+  allImages?: string,
+  imageWidth?: string
+  imageHeight?: string
+  uniqueName?: string
+}
+export const defaultPlaceholders: CommandContextPlaceholders = {
+  singleImagePrefix: '$$IMAGE_',
+  allImages: '$$ALLIMAGES',
+  imageWidth: '$$IMAGE_WIDTH',
```

Comment:
I think I missed your reply on why we are adding our own templating for things like width when it should be covered in imagemagick? http://www.imagemagick.org/script/escape.php (%w)